components/camel-stax/src/main/java/org/apache/camel/component/stax/StAXProcessor.java

```diff
@@ -49,25 +51,34 @@ public StAXProcessor(ContentHandler contentHandler) {
 
     @Override
     public void process(Exchange exchange) throws Exception {
-        InputSource is = exchange.getIn().getMandatoryBody(InputSource.class);
-        XMLStreamReader stream = exchange.getIn().getMandatoryBody(XMLStreamReader.class);
```

Comment:
This line also open a FileInputStream, which is not closed. 
The only way I find to close it is by casting the stream to the woodstox class (see finally clause).